webpack/assets/javascripts/react_app/components/ModelsTable/ModelsTableActions.js

```diff
@@ -0,0 +1,43 @@
+import {
+  TABLE_REQUEST,
+  TABLE_SUCCESS,
+  TABLE_FAILURE,
+  INIT_TABLE,
+} from './ModelsTableConsts';
+import { ajaxRequestAction } from '../../redux/actions/common';
+
+export const updateSearch = (current) => {
+  const init = {
+    page: current.params && current.params.page,
+    perPage: current.params && current.params.per_page,
+    order: current.params && current.params.order,
+    searchQuery: current.params && current.params.search,
+    results: current.table,
+  };
+
+  const payload = Object.keys(init)
+    .filter(k => ![undefined, null].includes(init[k]) && k !== 'order')
+    .reduce((acc, cur) => Object.assign({}, acc, { [cur]: init[cur] }), {});
+
+  const orderArray =
+    typeof init.order === 'string' ? init.order.split(' ') : [];
+  if (orderArray.length === 2) {
+    [payload.sortBy, payload.sortOrder] = orderArray;
+  }
+  payload.results = payload.results ? payload.results : [];
+
+  return {
+    type: INIT_TABLE,
+    payload,
+  };
+};
+
+export const sortTable = (url, controller) => dispatch =>
+  ajaxRequestAction({
+    dispatch,
+    requestAction: TABLE_REQUEST,
```

Comment:
If this action is responsible for sorting, wouldn't it be better to use more informative consts?
something like `SORT_HW_MODEL_TABLE_REQUEST` ?

Reply:
Super right! I actually found that `patternfly-react` as such a constant and I am going to use it here.